crates/router/src/routes/payment_methods.rs

```diff
@@ -223,6 +233,91 @@ pub async fn list_customer_payment_method_api_client(
     .await
 }
 
+/// Generate a form link for collecting payment methods for a customer
+#[utoipa::path(
+    post,
+    path = "/payment_methods/collect",
+    request_body=PaymentMethodCollectLinkRequest,
+    responses(
+        (status = 200, description = "Payment method collect link generated", body = PaymentMethodCollectLinkResponse),
+        (status = 400, description = "Missing Mandatory fields")
+    ),
+    tag = "Payment Methods Collect",
+    operation_id = "Generate form link for collecting payment methods for a customer",
+    security(("api_key" = []))
+)]
+#[instrument(skip_all, fields(flow = ?Flow::PaymentMethodCollectLink))]
+pub async fn initiate_pm_collect_link_flow(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    json_payload: web::Json<payment_methods::PaymentMethodCollectLinkRequest>,
+) -> HttpResponse {
+    let flow = Flow::PaymentMethodCollectLink;
+    Box::pin(api::server_wrap(
+        flow,
+        state,
+        &req,
+        json_payload.into_inner(),
+        |state, auth, req, _| {
+            payment_methods_routes::initiate_pm_collect_link(
+                state,
+                auth.merchant_account,
+                auth.key_store,
+                req,
+            )
+        },
+        &auth::ApiKeyAuth,
```

Comment:
just curious, wouldn't this be a dashboard call, later in future! if yes, then can we add JWT Auth

Reply:
Will do this in future, whenever needed